bucket.go

```diff
@@ -36,6 +36,7 @@ type Bucket struct {
 	Description         string        `json:"description"`
 	RetentionPolicyName string        `json:"rp,omitempty"` // This to support v1 sources
 	RetentionPeriod     time.Duration `json:"retentionPeriod"`
+	ShardGroupDuration  time.Duration `json:"shardGroupDuration"`
```

Comment:
We may need to make this:
```suggestion
	ShardGroupDuration  *time.Duration `json:"shardGroupDuration,omitempty"`
```
So Cloud can distinguish not-set from explicitly-0. Will get back to you after I work on the Cloud side of things...